builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -138,6 +137,10 @@ func resourceAwsAutoscalingGroupCreate(d *schema.ResourceData, meta interface{})
 		autoScalingGroupOpts.DefaultCooldown = aws.Integer(v.(int))
 	}
 
+	if d.HasChange("health_check") {
+		autoScalingGroupOpts.HealthCheckType = aws.String(d.Get("health_check_type").(string))
+	}
```

Comment:
I think this should be a `GetOk` and the check should be `ok && v.(string) != ""`